draco/synthesis/mockcatalog.py

```diff
@@ -795,20 +809,236 @@ def process(self, cat):
         # Generate standard normal z errors
         z_err = self.rng.normal(size=cat_z.shape[0])
         # Multiply by appropriate sigma
-        z_err *= (
-            self.sigma * np.ones_like(cat_z)
-            if self.sigma_type == "sigma_z"
-            else self.sigma * (1 + cat_z)
-        )
+        if self.use_catalog_z_errors:
+            if not np.any(cat_z_err):
+                self.log.error(
+                    "Warning: no existing z_error information in catalog, so no z errors will be added"
+                )
+            z_err *= cat_z_err
+        elif self.sigma_type == "sigma_z":
+            z_err *= self.sigma
+        else:  # self.sigma_type == "sigma_z_over_1plusz"
+            z_err *= self.sigma * (1 + cat_z)
 
         # Add errors to catalog redshifts
         cat_z += z_err
-        # Add errors in quadrature with existing stored errors
-        cat_z_err = (z_err ** 2 + cat_z_err ** 2) ** 0.5
+
+        # TODO: store information about error distribution in z_error field
 
         return cat
 
 
+class AddEBOSSZErrorsToCatalog(task.SingleTask, random.RandomTask):
+    """Add eBOSS-type random redshift errors to redshifts in a catalog.
+
+    See the docstrings for _{qso,elg,lrg}_velocity_error() for descriptions
+    of each redshift error distribution.
+
+    Note also that the errors are added to the catalog in place, such that
+    if the original catalog is subsequently used in a pipeline, it will
+    have the errors included.
+
+    Attributes
+    ----------
+    tracer : {"ELG"|"LRG"|"QSO"}
+        Generate redshift errors corresponding to this eBOSS sample.
+        If not specified, task will attempt to detect the tracre type from
+        the catalog's `tracer` attribute or its tag. Default: None
+    """
+
+    tracer = config.enum(["QSO", "ELG", "LRG"], default=None)
+
+    def process(self, cat):
+        """Generate random redshift errors and add to redshifts in catalog.
+
+        Parameters
+        ----------
+        cat : :class:`containers.SpectroscopicCatalog`
+            Input catalog.
+
+        Returns
+        ----------
+        cat_out : :class:`containers.SpectroscopicCatalog`
+            Catalog with redshift errors added.
+        """
+
+        tracer = self.tracer
+
+        # If tracer not specified in config, check to see whether it's stored
+        # in the catalog's 'tracer' attribute or in its tag
+        if tracer is None:
+            if "tracer" in cat.attrs:
+                tracer = cat.attrs["tracer"]
+            else:
+                for key in _velocity_error_function_lookup.keys():
+                    if key in cat.attrs["tag"]:
+                        tracer = key
+                        break
+
+                if tracer is None:
+                    raise ValueError(
+                        "Must specify eBOSS tracer in config property, "
+                        "catalog 'tracer' attribute, or catalog 'tag' attribute."
+                    )
```

Comment:
Maybe make this detection case insensitive? We've tended to use a bit of a mixed version of the tracer abbreviations.

Also check that the tracer attribute is actually understood.
```suggestion
        if tracer is None:
            if "tracer" in cat.attrs:
                tracer = cat.attrs["tracer"].upper()
                
                if tracer not in _velocity_error_function_lookup:
                    raise ValueError(
                        f"Tracer explicitly set to '{tracer}' in catalog, but value not supported."
                    )
            else:
                for key in _velocity_error_function_lookup.keys():
                    if key in cat.attrs["tag"].upper():
                        tracer = key
                        break

                if tracer is None:
                    raise ValueError(
                        "Must specify eBOSS tracer in config property, "
                        "catalog 'tracer' attribute, or catalog 'tag' attribute."
                    )
```